src/adagrams.js

```diff
@@ -1,15 +1,160 @@
+const pool = {
+  A: 9,
+  B: 2,
+  C: 2,
+  D: 4,
+  E: 12,
+  F: 2,
+  G: 3,
+  H: 2,
+  I: 9,
+  J: 1,
+  K: 1,
+  L: 4,
+  M: 2,
+  N: 6,
+  O: 8,
+  P: 2,
+  Q: 1,
+  R: 6,
+  S: 4,
+  T: 6,
+  U: 4,
+  V: 2,
+  W: 2,
+  X: 1,
+  Y: 2,
+  Z: 1,
+};
+
 export const drawLetters = () => {
-  // Implement this method for wave 1
+  const characters = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+  let hand = [];
+  let handFreq = {};
+
+  while (hand.length < 10) {
+    let randomInt = Math.floor(Math.random() * 26);
+    let randomLetter = characters.charAt(randomInt);
+
+    if (!hand.includes(randomLetter)) {
+      hand.push(randomLetter);
+      handFreq[randomLetter] = 1;
+    } else {
+      if (handFreq[randomLetter] < pool[randomLetter]) {
+        hand.push(randomLetter);
+        handFreq[randomLetter] += 1;
+      }
+    }
+  }
+
+  return hand;
 };
 
 export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
+  for (let char of input) {
+    if (lettersInHand.includes(char)) {
+      const index = lettersInHand.indexOf(char);
+      lettersInHand.splice(index, 1);
+    } else {
+      return false;
+    }
+  }
+
+  return true;
 };
 
 export const scoreWord = (word) => {
-  // Implement this method for wave 3
+  if (word.length === 0) {
+    return 0;
+  }
+  const wordUpper = word.toUpperCase();
+
+  const scoreChart = {
+    A: 1,
+    B: 3,
+    C: 3,
+    D: 2,
+    E: 1,
+    F: 4,
+    G: 2,
+    H: 4,
+    I: 1,
+    J: 8,
+    K: 5,
+    L: 1,
+    M: 3,
+    N: 1,
+    O: 1,
+    P: 3,
+    Q: 10,
+    R: 1,
+    S: 1,
+    T: 1,
+    U: 1,
+    V: 4,
+    W: 4,
+    X: 8,
+    Y: 4,
+    Z: 10,
+  };
+
+  let score = 0;
+
+  for (let letter of wordUpper) {
+    score += scoreChart[letter];
+  }
+
+  if (wordUpper.length > 6) {
+    score += 8;
+  }
+
+  return score;
+};
+
+export const highestScoreToDict = (highestWord, highestScore) => {
+  let highest = {
+    word: highestWord,
+    score: highestScore,
+  };
+
+  return highest;
 };
 
 export const highestScoreFrom = (words) => {
-  // Implement this method for wave 4
+  let highest = {};
+  let highestWord = [];
+  let highestScore = 0;
+
+  for (let word of words) {
+    if (highestWord.length === 0) {
+      highestWord.push(word);
+      highestScore = scoreWord(word);
```

Comment:
Instead of calling `scoreWord` multiple times, it would be more efficient to call it once and save the value in a variable that you can reuse.